### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: zephyr/src/supp_main.c [hunk 1]
```diff
@@ -89,6 +89,8 @@ static void iface_cb(struct net_if *iface, void *user_data)
 
 	/* TODO : make this user configurable*/
 	ifaces[0].ifname = "wlan0";
+
+	os_free(ifname);
```
Comment (sachinthegreen) [inline zephyr/src/supp_main.c:93]: We should have a check early on (line 83) to make sure that ifname is not NULL.

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Comment by rlubos** [comment]
@sachinthegreen @sr1dh48r @krish2718 Please prioritize the review process, if we want to get this in we need to do this ASAP.

File: zephyr/src/supp_main.c (hunk 1 above)
Comment (krish2718) [reply zephyr/src/supp_main.c:93]: os_free can take care of NULL pointer, but added a null check in the print.

**Review by sachinthegreen — APPROVED** [review]

**Comment by sachinthegreen** [comment]
> @sachinthegreen @sr1dh48r @krish2718 Please prioritize the review process, if we want to get this in we need to do this ASAP.

@rlubos : Approved from my end. @sr1dh48r is on vacation.

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
@krish2718 Can we merge and update SHA in https://github.com/nrfconnect/sdk-nrf/pull/8687?

**Pull request merged**